### PR TITLE
Creating Category Button

### PR DESCRIPTION
**Pull request by MaharshM**

### CODE REVIEW

File: app/Http/Controllers/CategoryController.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+        $category = Category::create([
+            'user_id'=> Auth::user()->id,
+            'name'=>$data['name'],
+            'description'=> $data['description']
+        ]);
+         return redirect('/category')->with('message', 'Category created successfully');
+    }
+
+    public function fetchCategories(){
```
Comment (abbasali) [inline app/Http/Controllers/CategoryController.php:32]: Why is this method created? This PR should only be for "Create Category" and not List categories. Remove it.

File: app/Http/Controllers/CategoryController.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Auth\Events\Validated;
+use App\Http\Requests\CreateCategoryRequest;
+use Illuminate\Contracts\Auth\Authenticatable;
+
+class CategoryController extends Controller
+{
+
+    public function create(){
```
Comment (abbasali) [inline app/Http/Controllers/CategoryController.php:17]: Read Ranium Guidelines and see how to format code. Use php-cs-fixer. There should be a space before `{`.

File: app/Http/Controllers/CategoryController.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    public function store(CreateCategoryRequest $request){
+        $data = $request->validated();
+
+        $category = Category::create([
+            'user_id'=> Auth::user()->id,
+            'name'=>$data['name'],
+            'description'=> $data['description']
+        ]);
+         return redirect('/category')->with('message', 'Category created successfully');
```
Comment (abbasali) [inline app/Http/Controllers/CategoryController.php:29]: Redirect using route name and not url. So this should be `redirect()->to('route-name')->...`

File: resources/views/category.blade.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<x-app-layout>
```
Comment (abbasali) [inline resources/views/category.blade.php:1]: This is the create form. So a better name for this view is `create.blade.php`. In fact, we should group all views of a model/controller in its own folder. So this file should be at `resources/views/category/create.blade.php`. All category related views will be in `category` folder.

File: resources/views/category.blade.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        <!-- This is an example component -->
+        <div class="py-9 max-w-2xl mx-auto flex justify-center">
+            <div class="bg-white shadow-md border border-gray-200 rounded-lg max-w-sm p-4 sm:p-6 lg:p-8 dark:bg-gray-800 dark:border-gray-700">
+                <x-success-status class="mb-4" :status="session('message')" />
+                <form class="space-y-6" action="{{ url('/add-category') }}" method="POST">
+                    @csrf
+                    <h3 class="text-xl font-medium text-gray-900 dark:text-white">Create Category</h3>
+                    <div>
+                        <x-label for="name" :value="__('Category Name')" />
+                        <x-input id="name" class="block mt-1 w-full" type="name" name="name" :value="old('name')" required autofocus />
```
Comment (abbasali) [inline resources/views/category.blade.php:15]: Where are we displaying validation errors for this field? Server side validation errors.

File: resources/views/components/success-status.blade.php [hunk 6]
```diff
@@ -0,0 +1,7 @@
+@props(['status'])
```
Comment (abbasali) [inline resources/views/components/success-status.blade.php:1]: Prop name should be `message` and not `status`. We are showing a success "message". Also the component name should be `success-message.blade.php`.

File: routes/web.php [hunk 7]
```diff
@@ -22,3 +23,7 @@
 })->middleware(['auth'])->name('dashboard');
 
 require __DIR__.'/auth.php';
+
+Route::get('/category', [CategoryController::class , 'create'])
```
Comment (abbasali) [inline routes/web.php:27]: The route URLs and names should be given as per [Resource Controllers](https://laravel.com/docs/9.x/controllers#actions-handled-by-resource-controller). In fact, you can use `Route::resource()` which will generate all URLs and routes.

**Review by abbasali — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/CategoryController.php (hunk 2 above)
Comment (MaharshM) [reply app/Http/Controllers/CategoryController.php:17]: done

File: app/Http/Controllers/CategoryController.php (hunk 1 above)
Comment (MaharshM) [reply app/Http/Controllers/CategoryController.php:32]: done.

File: app/Http/Controllers/CategoryController.php (hunk 3 above)
Comment (MaharshM) [reply app/Http/Controllers/CategoryController.php:29]: done.

File: resources/views/category.blade.php (hunk 4 above)
Comment (MaharshM) [reply resources/views/category.blade.php:1]: done.

File: resources/views/category.blade.php (hunk 5 above)
Comment (MaharshM) [reply resources/views/category.blade.php:15]: Created `CreateCategoryRequest.php` for that and mentioned `$data = $request->validated();` in `CategoryController`

File: resources/views/components/success-status.blade.php (hunk 6 above)
Comment (MaharshM) [reply resources/views/components/success-status.blade.php:1]: done

**Review by abbasali — APPROVED** [review]

**Pull request merged**